spec/requests/storage_services_spec.rb

```diff
@@ -79,4 +79,19 @@
 
     expect(response).to have_http_status(:ok)
   end
+
+  context 'Storage Services edit action' do
+    it "PUT /api/storage_services/:id'" do
+      require 'byebug'
+      byebug
+      provider = FactoryBot.create(:ems_autosde, :name => 'Autosde')
+      storage_service = FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService", :name => 'test_service1', :ext_management_system => provider)
```

Comment:
```suggestion
      storage_service = FactoryBot.create(:autosde_storage_manager, :ext_management_system => provider)
```

Reply:
extracted `provider` to `let()`.
extracting `storage_service` didn't quite work out with the `delete` and `get` specs.. 

in the `edit` spec, for some reason when I use `FactoryBot.create(:storage_service, ... )` the service does not support `update`, but it does when I use `FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService",...)`.

do u have any idea why? maybe it references the core repo `storage_service` model and not the provider's one?
it supposedly doesn't support `create` and `delete` as well, but those spec do pass..

```
    83:   context 'Storage Services edit action' do
    84:     it "PUT /api/storage_services/:id'" do
    85:       storage_service1 = FactoryBot.create("ManageIQ::Providers::Autosde::StorageManager::StorageService", :name => 'test_service1', :ext_management_system => provider)
    86:       storage_service2 = FactoryBot.create(:storage_service, :name => 'test_service1', :ext_management_system => provider)
    87: 
    88:       require 'byebug'
    89:       byebug
    90: 
=>  91:       api_basic_authorize('storage_service_edit')
    92:       put(api_storage_service_url(nil, storage_service))
    93: 
    94: 
    95:       expect(response.parsed_body["message"]).to include("Updating")
    96:       expect(response).to have_http_status(:ok)
    97:     end
    98:   end
    99: 
   100: end
(byebug) storage_service1.supports_update?
true
(byebug) storage_service2.supports_update?
false
(byebug) storage_service2.supports_create?
false
(byebug) storage_service2.supports_delete?
false
(byebug) storage_service1.supports_delete?
true

```

Reply:
sorry.

We tend to use `stub_supports` to ensure that you get the expected `supports?`.
In the far far future, we would like to change all these tests to use a generic providers with a bunch of expectations on them.